### PR TITLE
LPS-121908 Migrate management toolbar v2 usages in Sharing

### PR DESCRIPTION
**Pull request by ambrinchaudhary**
**Test plan:** 

1. Create a new user
2. Go to D&M and "Share" an entry with that user
3. Log in with the new user
4. From the user's menu (top right user's icon), select "Shared Content"

/cc @markocikos @julien @jonmak08

### CODE REVIEW

**Comment by ambrinchaudhary** [comment]
ci:test:sf

**Comment by ambrinchaudhary** [comment]
ci:test:relevant

File: modules/apps/sharing/sharing-web/src/main/resources/META-INF/resources/shared_assets/js/SharedAssetsManagementToolbarPropsTransformer.js [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+
+import {addParams, navigate, openSelectionModal} from 'frontend-js-web';
+
+export default function propsTransformer({
+	additionalProps: {selectAssetTypeURL, viewAssetTypeURL},
+	filterDropdownItems,
+	portletNamespace,
+	...otherProps
+}) {
+	Liferay.on('sharing:changed', () =>
```
Comment (julien) [inline modules/apps/sharing/sharing-web/src/main/resources/META-INF/resources/shared_assets/js/SharedAssetsManagementToolbarPropsTransformer.js:23]: Do we really want this in a propsTransformer?
Can't we put it somewhere else?

File: modules/apps/sharing/sharing-web/src/main/resources/META-INF/resources/shared_assets/js/SharedAssetsManagementToolbarPropsTransformer.js [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+		Liferay.Portlet.refresh('#p_p_id' + portletNamespace)
+	);
```
Comment (julien) [inline modules/apps/sharing/sharing-web/src/main/resources/META-INF/resources/shared_assets/js/SharedAssetsManagementToolbarPropsTransformer.js:25]: In any case we can use ``` `#p_p_id${portletNamespace}` ```

File: modules/apps/sharing/sharing-web/src/main/resources/META-INF/resources/shared_assets/js/SharedAssetsManagementToolbarPropsTransformer.js [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+
+	return {
+		...otherProps,
+		filterDropdownItems: filterDropdownItems.map((item) => {
+			return {
+				...item,
+				items: item.items.map((childItem) => {
+					return {
+						...childItem,
+						onClick() {
```
Comment (markocikos) [inline modules/apps/sharing/sharing-web/src/main/resources/META-INF/resources/shared_assets/js/SharedAssetsManagementToolbarPropsTransformer.js:35]: This code would work, but it's not consistent with component API.

We should create the same pattern here as for [actionDropdownItems](https://github.com/liferay/liferay-portal/blob/master/modules/apps/frontend-taglib/frontend-taglib-clay/src/main/resources/META-INF/resources/management_toolbar/ActionControls.js#L73-L89). With this, we would:
- declare `onClick` internaly 
- have a new callback, `onFilterDropdownItemClick`
- override that callback here in `propsTransformer`.

Note that we don't need to have `clone` or `dataAttributes` complications, as in the linked block. Just a call to the new callback.

**Review by markocikos — CHANGES_REQUESTED** [review]
@ambrinchaudhary  Left a few comments inline.

File: modules/apps/sharing/sharing-web/src/main/resources/META-INF/resources/shared_assets/js/SharedAssetsManagementToolbarPropsTransformer.js (hunk 3 above)
Comment (julien) [reply modules/apps/sharing/sharing-web/src/main/resources/META-INF/resources/shared_assets/js/SharedAssetsManagementToolbarPropsTransformer.js:35]: Hey @markocikos,

I've got the exact same use case for `document-library-web` so I'll work on that and hopefully submit it in short while.

**Comment by ambrinchaudhary** [comment]
Thank yoy @julien and @markocikos for your revision. Then, I will close this pull and will be sent again when Julien finishes the callback `onFilterDropdownItemClick`, and also will look where to correctly set the `Liferay.on('sharing:changed'... `

**Pull request closed without merging**